### PR TITLE
Add e2e test suite

### LINKED ISSUE #3: Update end-to-end testing to use cypress
From my point of view, [cypress](https://www.cypress.io/) seems to be a faster and more stable alternative to protractor when it comes to end-to-end testing of web applications. I'd like to suggest to switch to it and update the tests to cover all basic use-cases. The task list would be:

- [x] Integrate cypress test framework and drop protractor
- [x] Add the tests to the CI pipeline
- [x] Add all use-cases as tests by keeping further UI changes in mind

### PR DESCRIPTION
**Pull request by saschagrunert**
We now run cypress against a fixed set of assets, which will be only
included if directly specified via `npm run start:test`. This target
will be utilized by cypress to run the tests against it.

Closes #3

### CODE REVIEW

**Comment by saschagrunert** [comment]
The tests are running successfully in prow:

```
  Release Notes App

    ✓ should have the correct title (2280ms)

    ✓ should everything be visible (2115ms)

    ✓ should only entries for version '1.14.0' if slected (1288ms)

    ✓ should only entries for version '1.15.0' if slected (1340ms)

    ✓ should only contain 'kubelet' entries if selected (1242ms)

    ✓ should be possible to filter two options together (1238ms)

    ✓ should be possible to search entries (1224ms)

    ✓ should be possible to open the 'About' page (1255ms)
```

File: cypress/integration/app.spec.ts [hunk 1]
```diff
@@ -1,12 +1,167 @@
...
+    cy.get(option1140).should('be.visible');
+    cy.get(option1150).should('be.visible');
+    cy.get(optionsReleaseVersionsID).should('be.visible');
+    cy.get(searchBar).should('be.visible');
+    cy.location().should(loc => {
+      expect(loc.search).to.eq('');
+    });
+  });
+
+  it(`should show only entries for version '1.14.0' if slected`, () => {
```
Comment (jeefy) [inline cypress/integration/app.spec.ts]: slected -> selected

File: cypress/integration/app.spec.ts [hunk 2]
```diff
@@ -1,12 +1,167 @@
...
+      expect($c).to.contain(v1140entry3);
+      expect($c).to.contain(v1140entry4);
+      expect($c).to.contain(v1140entry5);
+    });
+    cy.location().should(loc => {
+      expect(loc.search).to.eq('?releaseVersions=1.14.0');
+    });
+  });
+
+  it(`should show only entries for version '1.15.0' if slected`, () => {
```
Comment (jeefy) [inline cypress/integration/app.spec.ts]: slected -> selected

File: cypress/integration/app.spec.ts (hunk 1 above)
Comment (saschagrunert) [reply cypress/integration/app.spec.ts]: Fixed :heavy_check_mark:

File: cypress/integration/app.spec.ts (hunk 2 above)
Comment (saschagrunert) [reply cypress/integration/app.spec.ts]: Fixed :heavy_check_mark:

**Comment by saschagrunert** [comment]
Rebased on top of the latest master branch to resolve the conflict.

**Comment by jeefy** [comment]
/approve
/lgtm
🎉

**Pull request merged**